### DISCUSSION TITLE
Multiple sessions/instances

### DISCUSSION THREAD

**Discussion by dionzand** [cpvalente/ontime#794] [Q&A]
I would like to use Ontime for a conference. But since we have multiple rooms/parallel sessions, what would be the best practice to run Ontime on the same server? Is it possible to create multiple sessions from the same instance? Or can you easily create multiple instances?

**Comment by cpvalente** [comment]
Hi @dionzand, thank you for reaching out.

The most efficient way to achieve multiple, simultaneous instances of Ontime would be using docker.
You should be able to run as many instances as you need and expose them from your server on different ports.

You can find the docker image over at dockerhub
https://hub.docker.com/r/getontime/ontime

Hope this helps

**Reply by dionzand** [reply]
Thanks for the quick reply! I will look into it 😃

**Reply by dionzand** [reply]
I got the docker instances to work :) thanks.
However, when I try to reach the instance from outside my LAN (I am serving the instances through NGINX), I loose functionality (greyed out buttons, and no running clock). Am I missing something or is Ontime not suited for public hosting? Thanks!

**Reply by cpvalente** [reply]
There are a few questions here, I will try to help:

> However, when I try to reach the instance from outside my LAN (I am serving the instances through NGINX), I loose functionality (greyed out buttons, and no running clock)

To my best knowledge, this setup works well, and several users have been doing this. It is tricky to know what might be the issue without more information. But I suspect that there is something wrong with the setup that makes that your UI cannot reach the server

The only functionality not available in docker is the changing of port. This is because we want you to change the port in the docker layer instead.
The grayed out buttons seem to indicate a lack of connection. You should be able to confirm this in the browser console

The Ontime server should be running on port 4001 (internally in the image). You can then expose it as another port to the outside of the container.

> or is Ontime not suited for public hosting

 Ontime is not secured, meaning it should only be hosted in a network you have control over.

Hope this helps

**Reply by dionzand** [reply]
Thanks again for the quick reply :) You were correct about the lack of connection.
Apparently I didnt configure the WebSocket correctly (https://www.nginx.com/blog/websocket-nginx/).

**Reply by cpvalente** [reply]
Great to hear you got setup.

The docker configuration has too many shapes for us to be able to document (or maybe even understand), but I realise that the documentation is lacking

Feel free to reach out if you have any feedback
